client/src/context/useDetailContext.tsx

```diff
@@ -0,0 +1,68 @@
+import { useState, useContext, createContext, FunctionComponent } from 'react';
+import { cardDetailItems } from '../components/CardDialog/initialDetailData';
+import { IDialogItem, IDialogItemContext } from '../context/types/detail';
+
+const initialDetailData: IDialogItemContext = {
+  items: [] as IDialogItem[],
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  addItem: (_item: IDialogItem) => false,
+  removeItem: (_itemId: string) => false,
+  resetItems: () => false,
+};
+
+export const DialogContext = createContext<IDialogItemContext>(initialDetailData);
+
+export const DialogProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [items, setItems] = useState<Array<IDialogItem>>(cardDetailItems);
+
+  const swapItems = (items: IDialogItem[], source: string, destination: string, itemId: string): IDialogItem[] => {
+    const itemsCopy = [...items];
+    const itemIndex = itemsCopy.findIndex((item: IDialogItem) => item.id === itemId);
+    if (itemIndex > -1) {
+      //    Will need some logic for swapping items
+      //   const [item] = itemsCopy.splice(source.index, 1);
+      //   itemsCopy.splice(destination.index, 0, item);
+      return itemsCopy;
+    }
+    return items;
+  };
+
+  const resetItems = (): boolean => {
+    setItems(cardDetailItems);
+    return false;
+  };
+
+  const addItem = (item: IDialogItem): boolean => {
+    const itemsPlusOne = [...items, item];
+    setItems(itemsPlusOne);
+    return false;
+  };
+
+  const removeItem = (itemId: string): boolean => {
+    const remaining = items.filter((item) => item.id !== itemId);
+    console.log(remaining);
+    setItems(remaining);
+    return false;
+  };
```

Comment:
Not sure why these are returning false, perhaps true would be more suitable? In the future, we would want to return true or false if the board update is successful 👍🏻

Reply:
They're returning false because I was getting typescript errors when they were void and couldn't figure it out haha. Honestly I wasn't even thinking about using a boolean return to check if it was successful. I can switch to void or make them return true by default, either way

Reply:
If you are wanting to do a void response just enter `return;` and it should not give you a problem 👍🏻